graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -182,6 +133,17 @@ def freeze(self):
         self.in_connections = frozenset(self.in_connections)
         self.out_connections = frozenset(self.out_connections)
 
+    def validate_endpoints(self):
```

Comment:
All the other validation functions are not defined on instances of objects, and are just defined at module level. Let's stick to that paradigm to avoid confusing the reader unless there is very strong reason not to do so.